### ISSUE TITLE
Move 1 vs 2 plot city radius into GameOptions instead of compile time flag

### ISSUE THREAD

**Issue by Nightinggale** [We-the-People-civ4col-mod/Mod#60]
People can't agree if 1 or 2 plot radius is the right option, which is why we have ended up with two versions. The problem is that it requires two DLL files. Python has code for both cases and has a bool check to see which kind of DLL file it works with.

If we can move the city radius selection from compile time to GameOptions, we will remove the need to distribute two DLL files.

**Comment by Nightinggale** [comment]
This should likely be a map setting rather than a GameOption. The difference is that the map settings are asked in both custom games and "normal" games while GameOptions are asked in custom games only while the "normal" games use default settings for GameOptions.

**Comment by Nightinggale** [comment]
I started working on this. I decided on the design where the first game option is the catchment radius and that it should be that for all maps. Sure it means we have to update all map generators to match that, but that's doable.

By making sure it's always at index 0, it will be possible to hook into the code whenever the settings change (including loading a game) even without knowing the current map script. This makes it possible to inform CvGlobals whenever the setting change.

CvGlobals currently contains the define stuff used to make the two dlls different. This means since CvGlobals knows the current value at all time, it should be able to switch the "constants" at runtime.
